### ISSUE TITLE
UI will permanently break if bad data is loaded

### ISSUE THREAD

**Issue by sman591** [sman591/rit-class-stats#11]
Redux state is cached to local storage for fast re-visit loads. However, this means that if bad data was cached to local storage, this bad data is immediately loaded on the next visit before any new API requests go out. If the bad data causes a JavaScript exception, the code fails to make new API calls, and the visitor is permanently locked at an old, broken state.

A few good measures:
* [ ] Catch errors and/or utilize React's new error boundaries to prevent errors like this from stalling the entire app
* [ ] Don't utilize cached local storage if older than a few hours or days

**Comment by audiolion** [comment]
Dealing with cache invalidation is the primary reason I am not using redux-persist/service workers, it is really frustrating to deal with and get it right. Kudos to using it here though its a good opportunity to learn it.

**Comment by sman591** [comment]
Yeah, figured this project would be a simple test bed to try out with. Ultimately if it becomes a problem I'll just turn it off
